Add API docs generation

SemyonSinchenko opened the pull request:
On branch api-docs-generation
 Changes to be committed:
	modified:   .github/workflows/website.yml
	modified:   .gitignore
	new file:   docs/gen_ref_pages.py
	modified:   mkdocs.yml
	modified:   poetry.lock
	modified:   pyproject.toml


Close #83


This PR includes the following changes:
1. Update dependencies as discussed [here](https://github.com/pyspark-ai/pyspark-ai/issues/83#issuecomment-1742064246)
2. Add generation file as discussed in the same comment
3. Update mkdocs.yaml
4. **Update site-workflow by including all the dependencies** (see commentary in the code)

I made a deployment from my fork and it looks like this:
https://semyonsinchenko.github.io/pyspark-ai/reference/SUMMARY/

SemyonSinchenko committed:
Add API docs generation

 On branch api-docs-generation
 Changes to be committed:
	modified:   .github/workflows/website.yml
	modified:   .gitignore
	new file:   docs/gen_ref_pages.py
	modified:   mkdocs.yml
	modified:   poetry.lock
	modified:   pyproject.toml

SemyonSinchenko commented:
@gengliangwang This PR is ready for review. Please, make a look, when you have time! Tnx in advance!

gengliangwang commented:
@SemyonSinchenko Thanks for the work. However, the API references from https://semyonsinchenko.github.io/pyspark-ai/reference/SUMMARY/ are not the ones we expect to show to users. We only want to expose the APIs with the `df.ai.` prefix.

SemyonSinchenko committed:
Try to filter part of blocks

 On branch api-docs-generation
 Changes to be committed:
	modified:   docs/gen_ref_pages.py

SemyonSinchenko commented:
@gengliangwang I can reduce the scope of exposed API. You may check the updated link: https://semyonsinchenko.github.io/pyspark-ai/reference/SUMMARY/

But `df.ai` is a runtime magic, I don't think we can 100% automate it...

Btw, do I understand correctly, that you want to see in API docs methods in form `DataFrame.ai.XXX`? Or it is enough to expose the actual classes (SparkAI, AIUtils, etc.) but limit the scope?

gengliangwang commented:
> you want to see in API docs methods in form DataFrame.ai.XXX?

Yes, we only want users to use `df.ai...`. Exposing other methods will confuse users. 

Thanks for updating https://semyonsinchenko.github.io/pyspark-ai/reference/SUMMARY/. But it seems making the doc a bit more complicated.
